packages/expo-cli/lib/insert.js

```diff
@@ -8,19 +8,31 @@ const importRe = /from ["']@bugsnag\/expo["']/
 const requireRe = /require\(["']@bugsnag\/expo["']\)/
 
 module.exports = async (projectRoot) => {
-  try {
-    const appJsPath = join(projectRoot, 'App.js')
-    const appJs = await promisify(readFile)(appJsPath, 'utf8')
-    if (importRe.test(appJs) || requireRe.test(appJs)) {
-      return '@bugsnag/expo is already imported in App.js'
+  // check if app.ts exists. If it does exist, check for/insert bugsnag in app.ts
+  const appTsPath = join(projectRoot, 'App.ts')
+  if (existsSync(appTsPath)) {
+    const appTs = await promisify(readFile)(appTsPath, 'utf8')
+    if (importRe.test(appTs) || requireRe.test(appTs)) {
+      return '@bugsnag/expo is already imported in App.ts'
     }
-    await promisify(writeFile)(appJsPath, `${await getCode(projectRoot)}\n${appJs}`, 'utf8')
-  } catch (e) {
-    // swallow and rethrow for errors that we can produce better messaging for
-    if (e.code === 'ENOENT') {
-      throw new Error(`Couldn’t find App.js in "${projectRoot}". Is this the root of your Expo project?`)
+    await promisify(writeFile)(appTsPath, `${await getCode(projectRoot)}\n${appTs}`, 'utf8')
+
```

Comment:
You can `return` here and bail out of the function, and remove the nested `else` clause below. Not a massive concern here, but a very useful pattern for when you get a lot of nesting.